controllers/baremetalset_controller.go

```diff
@@ -19,8 +19,11 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"net"
+	"strings"
 	"time"
 
+	"github.com/apparentlymart/go-cidr/cidr"
```

Comment:
I hadn't been using this for my IP/CIDR code. But it tidies things up and it looks like some other dependencies use it as well so lets add it in